lua/tardis/libraries/libraries/sh_debug_portals.lua

```diff
@@ -0,0 +1,316 @@
+
+            row2:Setup( "Generic" )
+            row2:SetValue(value)
+            row2.DataChanged = function( _, val )
+                if tonumber(val) == nil then return end
+                row1:SetValue(val)
+                update_func(val)
+            end
+
+            return row1, row2
+        end
+
+
+        local x, x2 = SetupProperty("Position", "X", px, 100, function(val)
+            px = val
+            UpdatePortalPos()
+        end)
+        local y, y2 = SetupProperty("Position", "Y", py, 100, function(val)
+            py = val
+            UpdatePortalPos()
+        end)
+        local z, z2 = SetupProperty("Position", "Z", pz, 100, function(val)
+            pz = val
+            UpdatePortalPos()
+        end)
+
+        local ap, ap2 = SetupProperty( "Angle", "Pitch", ang_p, 360, function(val)
+            ang_p = val
+            UpdatePortalAng()
+        end)
+        local ay, ay2 = SetupProperty( "Angle", "Yaw", ang_y, 360, function(val)
+            ang_y = val
+            UpdatePortalAng()
+        end)
+        local ar, ar2 = SetupProperty( "Angle", "Roll", ang_r, 360, function(val)
+            ang_r = val
+            UpdatePortalAng()
+        end)
+
+        local w, w2 = SetupProperty("Size", "Width", width, 0, 300, function(val)
+            width = val
+            UpdatePortalSize()
+        end)
+        local h, h2 = SetupProperty("Size", "Height", height, 0, 300, function(val)
+            height = val
+            UpdatePortalSize()
+        end)
+
+        local thick, thick2 = SetupProperty( "3D", "Thickness", thickness, 150, function(val)
+            thickness = val
+            UpdatePortal3D()
+        end)
+
+        local inv = pr:CreateRow( "3D", "Inverted" )
+        inv:Setup( "Bool" )
+        inv:SetValue(inverted)
+        inv.DataChanged = function( _, val )
+            inverted = val
+            UpdatePortal3D()
+        end
+
+
+        local epox, epox2 = SetupProperty("Exit Point", "X", epo_x, 300, function(val)
+            epo_x = val
+            UpdatePortalExitOffset()
+        end)
+        local epoy, epoy2 = SetupProperty("Exit Point", "Y", epo_y, 300, function(val)
+            epo_y = val
+            UpdatePortalExitOffset()
+        end)
+        local epoz, epoz2 = SetupProperty("Exit Point", "Z", epo_z, 300, function(val)
+            epo_z = val
+            UpdatePortalExitOffset()
+        end)
+
+        local eaop, eaop2 = SetupProperty( "Exit Point", "Pitch", eao_p, 360, function(val)
+            eao_p = val
+            UpdatePortalExitOffset()
+        end)
+        local eaoy, eaoy2 = SetupProperty( "Exit Point", "Yaw", eao_y, 360, function(val)
+            eao_y = val
+            UpdatePortalExitOffset()
+        end)
+        local eaor, eaor2 = SetupProperty( "Exit Point", "Roll", eao_r, 360, function(val)
+            eao_r = val
+            UpdatePortalExitOffset()
+        end)
+
+
+        local ep_category = pr:GetCategory("Exit Point")
```

Comment:
I think clarifying this is an offset for the exit point a bit better might be useful, I was a bit confused as to what this did for a while